>From b721166ad224c882a23911b3d057a7ce636f00a3 Mon Sep 17 00:00:00 2001
From: Salvatore Bonaccorso <carnil@debian.org>
Date: Thu, 13 Jun 2013 20:02:04 +0200
Subject: [PATCH] Adapt to changed output for git status in newer git versions

Newer versions of git emit "nothing to commit, working directory clean"
instead of "nothing to commit (working directory clean)" on git status
when a working directory is clean.

Adjust the parsing to match for both cases, as git commit would exit
with nonzero status and thus debcommit exit.
---
 scripts/debcommit.pl |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/scripts/debcommit.pl b/scripts/debcommit.pl
index 675d063..f129f38 100755
--- a/scripts/debcommit.pl
+++ b/scripts/debcommit.pl
@@ -520,7 +520,7 @@ sub commit {
 	    # check to see if the WC is clean. git-commit would exit
 	    # nonzero, so don't run it in --all or --release mode.
 	    my $status=`LANG=C git status`;
-	    if ($status=~/nothing to commit \(working directory clean\)/) {
+	    if ($status =~ /nothing to commit/ && $status =~ /working directory clean/) {
 		    print $status;
 		    return;
 	    }
-- 
1.7.10.4

